### DISCUSSION TITLE
Optionally expose current Task in titlebar

### DISCUSSION THREAD

**Discussion by a13ph** [minbrowser/min#2340] [Ideas]
For time tracking purposes, I'd like to have current task (workspace / tab group) name to be show in titlebar, so that timetracking and some other apps could read the title and do their magic based on that.

Right now, I'm trying to get current task through json file with session (tabs) information. 
It seems that session json is not updated on task switch through task switcher (when using only mouse for example).  
(I'll get to the reporting that bug properly through bug template one day I hope)
Relying on json file is of limited use anyways, as not all software can be integrated this way and some rely on window titles.

**Comment by PalmerAL** [comment]
Makes sense to me. I think it would be fine to do this by default actually, since the title bar is hidden in most cases anyway - is there a situation in which you'd want to turn this off?

> It seems that session json is not updated on task switch through task switcher (when using only mouse for example).

Most likely, what you're seeing is that the session restore file only gets written to disk every 30sec to avoid putting too much load on your drive; if you wait a bit the changes should show up.

**Reply by a13ph** [reply]
Me personally, I'd always have it ON.

But I can see possible counterpoints like:
- slight increase in attack surface (not all tracking software is harmless)
- someone might already have a workflow (e.g. automation) relying on window name to be exact string "Min"

Re: 30 secs, my thanks, I'll take that into account!

**Reply by PalmerAL** [reply]
> slight increase in attack surface (not all tracking software is harmless)

We generally don't try to protect against local attacks - it's not really possible, given that nothing is sandboxed from each other in desktop OSs.

> someone might already have a workflow (e.g. automation) relying on window name to be exact string "Min"

I imagine these could be updated pretty easily.

So I'm OK with just implementing this, and we can go back and add a preference if anyone reports problems with it. Thanks for bringing those up though.

**Comment by PalmerAL** [comment]
Implemented in 02cb14248f4215a978bb1deda3c15f6a50150f38.
